tidy3d/web/api/mode.py

```diff
@@ -319,7 +320,20 @@ def create(
         folder = Folder.get(folder_name, create=True)
 
         mode_solver.validate_pre_upload()
-        mode_solver.simulation.validate_pre_upload(source_required=False)
+        if isinstance(mode_solver.simulation, Simulation):
+            mode_solver.simulation.validate_pre_upload(source_required=False)
+        elif isinstance(mode_solver.simulation, EMESimulation):
+            # TODO: support this web api in the future
+            raise SetupError(
+                "'EMESimulation' is not yet supported in the "
+                "remote mode solver web api. Please construct a 'Simulation' "
+                "with the same 'medium', 'structures', 'grid_spec', 'symmetry', etc. as the "
+                "'EMESimulation' in order to use the remote mode solver. "
+                "Alternatively, you can add a 'ModeSolverMonitor' to the 'EMESimulation'."
+            )
+            # mode_solver.simulation.validate_pre_upload()
+        else:
+            raise SetupError("Simulation type not supported in the remote mode solver web api.")
```

Comment:
this would never be raised based on the type of `simulation`? but I guess can;'t hurt to have? same comment in the sim_data one.

Reply:
Yeah, this is mostly for down the line when we add mode types, in particular `ModeSimulation`, just to make sure nothing slips through the cracks.